training/service.go

```diff
@@ -0,0 +1,366 @@
+	}
+	data.Status = string(response.Status)
+	err = service.storage.Put(key, data)
+	return
+}
+
+func (service ModelService) updateModelDetails(request *UpdateModelRequest, response *ModelDetailsResponse) (err error) {
+	key := service.getModelKeyToUpdate(request.ModelDetailsRequest)
+	if data, err := service.getModelDataForUpdate(request, response); err != nil {
+		if data, ok, err := service.storage.Get(key); err != nil && ok {
+			data.AuthorizedAddresses = request.AddressList
+			data.isPublic = request.IsPubliclyAccessible
+			data.UpdatedByAddress = request.ModelDetailsRequest.Authorization.SignerAddress
+			data.Status = string(response.Status)
+		}
+
+		err = service.storage.Put(key, data)
+	}
+	return
+}
+
+func (service ModelService) updateModelDetailsForStatus(request *ModelDetailsRequest, response *ModelDetailsResponse) (err error) {
+	key := service.getModelKeyToUpdate(request)
+	if data, err := service.getModelDataForStatusUpdate(request, response); err != nil {
+		err = service.storage.Put(key, data)
+	}
+	return
+}
+func (service ModelService) getModelKeyToCreate(request *CreateModelRequest, response *ModelDetailsResponse) (key *ModelUserKey) {
+	key = &ModelUserKey{
+		OrganizationId: config.GetString(config.OrganizationId),
+		ServiceId:      config.GetString(config.ServiceId),
+		GroupId:        service.organizationMetaData.GetGroupIdString(),
+		MethodName:     request.MethodName,
+		ModelId:        response.ModelDetails.ModelId,
+	}
+	return
+}
+
+func (service ModelService) getModelKeyToUpdate(request *ModelDetailsRequest) (key *ModelUserKey) {
+	key = &ModelUserKey{
+		OrganizationId: config.GetString(config.OrganizationId),
+		ServiceId:      config.GetString(config.ServiceId),
+		GroupId:        service.organizationMetaData.GetGroupIdString(),
+		MethodName:     request.ModelDetails.MethodName,
+		ModelId:        request.ModelDetails.ModelId,
+	}
+	return
+}
+
+func (service ModelService) getModelDataForUpdate(request *UpdateModelRequest, response *ModelDetailsResponse) (data *ModelUserData, err error) {
+	data, err = service.getModelDataForStatusUpdate(request.ModelDetailsRequest, response)
+	return
+}
+
+func (service ModelService) getModelDataForStatusUpdate(request *ModelDetailsRequest, response *ModelDetailsResponse) (data *ModelUserData, err error) {
+	key := service.getModelKeyToUpdate(request)
+	ok := false
+
+	if data, ok, err = service.storage.Get(key); err != nil && !ok {
+		log.WithError(fmt.Errorf("issue with retrieving model data from storage"))
+	}
+	return
+}
+func (service ModelService) GetAllModels(c context.Context, request *AccessibleModelsRequest) (response *AccessibleModelsResponse, err error) {
+	if request == nil || request.Authorization == nil {
+		return &AccessibleModelsResponse{Status: Status_ERROR},
+			fmt.Errorf(" INVALID  REQUEST DETAILS , no Authorization provided ")
+	}
+	if err = service.verifySignerForCreateModel(request.Authorization); err != nil {
```

Comment:
Why are we calling "verifySignerForCreateModel" - we are not creating a model here correct? if "verifySignerForCreateModel" is the generic method that checks if caller has access to the model, lets give it a generic name such as checkModelAccess or anything else better

Reply:
signture comprises of 
methodname, signer_address,blocknumber 
Any method called needs to be signed ,verify*methods are for signture verification , will rename it to verifySignature..rather than verifySigner